libs/ml/examples/cbow_simple/polyfill.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "math/tensor.hpp"
+#include "meta/is_iterable.hpp"
+
+namespace fetch {
+namespace math {
+
+template <typename T1, typename T2>
+fetch::meta::IsIterableTwoArg<T1, T2, void> PolyfillInlineAdd(T1 &ret, T2 const &other)
+{
```

Comment:
this looks like vectorised code to add a tensor slice by slice to another tensor. Would be nice to have a comment that says this is what it does. I may be wrong but i'm not aware of polyfill being a standard name for this operation (I don't see it in numpy and wiki's entry on polyfill(programming) is unrelated), so I would say a comment is worthwhile